app/scripts/lib/multichain-api/provider-authorize/helpers.ts

```diff
@@ -0,0 +1,61 @@
+import { CaipAccountId, Hex } from '@metamask/utils';
+import {
+  NetworkClientId,
+  NetworkController,
+} from '@metamask/network-controller';
+import { ScopesObject, validateScopedPropertyEip3085 } from '../scope';
+
+export const assignAccountsToScopes = (
+  scopes: ScopesObject,
+  accounts: Hex[],
+) => {
+  Object.keys(scopes).forEach((scope) => {
+    if (scope !== 'wallet') {
+      scopes[scope].accounts = accounts.map(
+        (account) => `${scope}:${account}` as unknown as CaipAccountId, // do we need checks here?
+      );
+    }
+  });
+};
+
+export const validateAndUpsertEip3085 = async ({
+  scopeString,
+  eip3085Params,
+  origin,
+  upsertNetworkConfiguration,
+  findNetworkClientIdByChainId,
+}: {
+  scopeString: string;
+  eip3085Params: unknown;
+  origin: string;
+  upsertNetworkConfiguration: NetworkController['upsertNetworkConfiguration'];
+  findNetworkClientIdByChainId: NetworkController['findNetworkClientIdByChainId'];
+}): Promise<undefined | NetworkClientId> => {
+  const validParams = validateScopedPropertyEip3085(scopeString, eip3085Params);
```

Comment:
same with this? As mentioned before, this data will not be mutable by the request so if we validate it before sending it to the approval controller we shouldn't need to validate it again afterwards.

Reply:
yeah i suppose that's true. We only need
`const validParams = validateAddEthereumChainParams(eip3085Params);` at this point to make the upsert call

Reply:
done here https://github.com/MetaMask/metamask-extension/pull/25873/commits/f765e37eb25f3882e709b7990d0a22326188552c